cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/CassandraTablesCreator.java

```diff
@@ -1,343 +1,394 @@
+				.withClusteringColumn(START_DATE, DataTypes.DATE)
+				.withClusteringOrder(START_DATE, ClusteringOrder.ASC);
+		
+		exec.executeQuery(create.asCql(), true);
+		logger.info("Table '{}' has been created", tableName);
+	}
+	
+	public void createTestEventsMessagesTable() throws IOException
+	{
+		String tableName = settings.getTestEventsMessagesTableName();
+		if (isTableExists(tableName))
+			return;
+		
+		CreateTable create = SchemaBuilder.createTable(settings.getKeyspace(), tableName).ifNotExists()
+				.withPartitionKey(INSTANCE_ID, DataTypes.UUID)
+				.withPartitionKey(TEST_EVENT_ID, DataTypes.TEXT)
+				.withClusteringColumn(ID, DataTypes.TIMEUUID)
+				.withColumn(MESSAGE_IDS, DataTypes.setOf(DataTypes.TEXT));
+//				.withClusteringColumn(MESSAGE_IDS, DataTypes.frozenSetOf(DataTypes.TEXT));
+		
+		exec.executeQuery(create.asCql(), true);
+		logger.info("Table '{}' has been created", tableName);
+	}
+	
+	public void createMessagesTestEventsTable() throws IOException
+	{
+		String tableName = settings.getMessagesTestEventsTableName();
+		if (isTableExists(tableName))
+			return;
+		
+		CreateTable create = SchemaBuilder.createTable(settings.getKeyspace(), tableName).ifNotExists()
+				.withPartitionKey(INSTANCE_ID, DataTypes.UUID)
+				.withPartitionKey(MESSAGE_ID, DataTypes.TEXT)
+				.withClusteringColumn(TEST_EVENT_ID, DataTypes.TEXT)
+				.withColumn(BATCH_ID, DataTypes.TEXT);
+		
+		exec.executeQuery(create.asCql(), true);
+		logger.info("Table '{}' has been created", tableName);
+	}
+	
+	protected void createMessagesTable(String name) throws IOException
+	{
+		if (isTableExists(name))
+			return;
+		
+		CreateTableWithOptions create = SchemaBuilder.createTable(settings.getKeyspace(), name).ifNotExists()
+				.withPartitionKey(INSTANCE_ID, DataTypes.UUID)
+				.withPartitionKey(STREAM_NAME, DataTypes.TEXT)
+				.withClusteringColumn(DIRECTION, DataTypes.TEXT)
+				.withClusteringColumn(MESSAGE_INDEX, DataTypes.BIGINT)
+				.withColumn(STORED_DATE, DataTypes.DATE)
+				.withColumn(STORED_TIME, DataTypes.TIME)
+				.withColumn(FIRST_MESSAGE_DATE, DataTypes.DATE)
+				.withColumn(FIRST_MESSAGE_TIME, DataTypes.TIME)
+				.withColumn(LAST_MESSAGE_DATE, DataTypes.DATE)
+				.withColumn(LAST_MESSAGE_TIME, DataTypes.TIME)
+				.withColumn(COMPRESSED, DataTypes.BOOLEAN)
+				.withColumn(CONTENT, DataTypes.BLOB)
+				.withColumn(MESSAGE_COUNT, DataTypes.INT)
+				.withColumn(LAST_MESSAGE_INDEX, DataTypes.BIGINT)
+				.withClusteringOrder(DIRECTION, ClusteringOrder.ASC)
+				.withClusteringOrder(MESSAGE_INDEX, ClusteringOrder.ASC);
+		
+		exec.executeQuery(create.asCql(), true);
+		logger.info("Table '{}' has been created", name);
+	}
+
+	private void createStreamsTable(String name) throws IOException
+	{
+		CreateTable create = SchemaBuilder.createTable(settings.getKeyspace(), name).ifNotExists()
+				.withPartitionKey(INSTANCE_ID, DataTypes.UUID)
+				.withClusteringColumn(STREAM_NAME, DataTypes.TEXT);
```

Comment:
With this definition the partition can grow indefinitely: as long as streams are added, the partition will grow and we can't stop it. Maybe it is better to partition by both instance_id and stream_name,